mongomock/filtering.py

```diff
@@ -350,6 +355,8 @@ def _get_compare_type(val):
         return 45
     if isinstance(val, _RE_TYPES):
         return 50
+    if isinstance(val, DBRef):
+        return 55
```

Comment:
Thanks for adding it here as well, please make sure that the sorting of type is working as in `pymongo` by adding a line in `test__sort_mixed_types` in `test__mongomock.py`